### PR TITLE
Bug Fix Pack - 1-25-2019

### PR DESCRIPTION
**Pull request by Surrealaser**
:cl: Surrealistik
fix: Fixed sentries not spawning with ammo.
fix: 'Dumb' fire sentries are no longer connected to the camera network.
fix: Supply drop pads fixed to work with lasgun power pack rechargers.
fix: Misc Ravager rage fixes.
fix: Fixed runtimes for projectiles.
fix: Assault shotgun uses specialist skill.
fix: Movement during pounces causing wonky/unintended behaviour.
tweak: Minor sentry alert tweaks.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/updated_projectiles/projectile.dm [hunk 1]
```diff
@@ -1,8 +1,8 @@
 
 //Some debug variables. Toggle them to 1 in order to see the related debug messages. Helpful when testing out formulas.
-#define DEBUG_HIT_CHANCE	0
-#define DEBUG_HUMAN_DEFENSE	0
-#define DEBUG_XENO_DEFENSE	0
+#define DEBUG_HIT_CHANCE	1
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/projectile.dm]:
```suggestion
#define DEBUG_HIT_CHANCE	FALSE
```

File: code/modules/projectiles/updated_projectiles/projectile.dm [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
+#define DEBUG_HUMAN_DEFENSE	1
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/projectile.dm]:
```suggestion
#define DEBUG_HUMAN_DEFENSE	FALSE
```

File: code/modules/projectiles/updated_projectiles/projectile.dm [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
+#define DEBUG_XENO_DEFENSE	1
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/projectile.dm]:
```suggestion
#define DEBUG_XENO_DEFENSE	FALSE
```

File: code/modules/cm_marines/equipment/sentries.dm [hunk 4]
```diff
@@ -278,16 +278,15 @@
...
 	spark_system = new /datum/effect_system/spark_spread
 	spark_system.set_up(5, 0, src)
 	spark_system.attach(src)
 	cell = new /obj/item/cell/high(src)
 	camera = new (src)
 	camera.network = list("military")
 	camera.c_tag = "[name] ([rand(0, 1000)])"
-	spawn(2)
-		stat = 0
+	stat = 0
```
Comment (Rohesie) [inline code/modules/cm_marines/equipment/sentries.dm]:
```suggestion
	stat = NOFLAGS
```

File: code/modules/cm_marines/equipment/sentries.dm [hunk 5]
```diff
@@ -1141,33 +1140,39 @@
...
 	burst_fire = TRUE
-	rounds = 50000
 	rounds_max = 50000
 	icon_state = "sentry_base"
 
-/obj/machinery/marine_turret/premade/New()
-	spark_system = new /datum/effect_system/spark_spread
-	spark_system.set_up(5, 0, src)
-	spark_system.attach(src)
+/obj/machinery/marine_turret/premade/Initialize()
```
Comment (Rohesie) [inline code/modules/cm_marines/equipment/sentries.dm]: If you change it to `Initialize()` and there's a `	. = ..()` you need to go down that well and change the `New()` that get called there, and also go up and change the `New()` procs that may have been calling this one.

Comment (Surrealaser) [reply code/modules/cm_marines/equipment/sentries.dm]: I believe this has been handled

Comment (Rohesie) [reply code/modules/cm_marines/equipment/sentries.dm]: Are you sure all the children and parents of this proc are now `Initialize()`?

File: code/modules/projectiles/updated_projectiles/projectile.dm [hunk 6]
```diff
@@ -657,9 +664,10 @@ Normal range for a defender's bullet resist should be something around 30-50. ~N
 			#endif
 
 		var/penetration = P.ammo.penetration > 0 || armor > 0 ? P.ammo.penetration : 0
-		if(src == P.shot_from.sniper_target(src))
-			damage *= SNIPER_LASER_DAMAGE_MULTIPLIER //+50% damage vs the aimed target
-			penetration *= SNIPER_LASER_ARMOR_MULTIPLIER //+50% penetration vs the aimed target
+		if(P.shot_from) //Runtimes bad
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/projectile.dm]:
```
		if(P.shot_from && src == P.shot_from.sniper_target(src)) //Runtimes bad
```

File: code/modules/projectiles/updated_projectiles/projectile.dm [hunk 7]
```diff
@@ -542,9 +548,10 @@ Normal range for a defender's bullet resist should be something around 30-50. ~N
 		to_chat(world, "<span class='debuginfo'>Initial armor is: <b>[armor]</b></span>")
 		#endif
 		var/penetration = P.ammo.penetration > 0 || armor > 0 ? P.ammo.penetration : 0
-		if(src == P.shot_from.sniper_target(src))
-			damage *= SNIPER_LASER_DAMAGE_MULTIPLIER //+50% damage vs the aimed target
-			penetration *= SNIPER_LASER_ARMOR_MULTIPLIER //+50% penetration vs the aimed target
+		if(P.shot_from) //Runtimes bad
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/projectile.dm]: Same.

File: code/modules/projectiles/updated_projectiles/projectile.dm [hunk 8]
```diff
@@ -251,13 +251,19 @@
...
-						to_chat(world, "Hit Chance 2: [hit_chance]")
-						#endif
+					if(hit_roll < 25) //Sniper targets more likely to hit
+						if(shot_from) //Avoid sentry run times
+							if(!shot_from.sniper_target(A))
+								def_zone = pick(base_miss_chance)	// Still hit but now we might hit the wrong body part
+						else
+							def_zone = pick(base_miss_chance)	// Still hit but now we might hit the wrong body part
+
+					if(shot_from) //Avoid sentry run times
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/projectile.dm]:
```
					if(shot_from && !shot_from.sniper_target(A)) //Avoid sentry run times
```

**Comment by Rohesie** [comment]
One change left before merging.

File: code/modules/projectiles/updated_projectiles/projectile.dm [hunk 9]
```diff
@@ -249,14 +249,18 @@
...
+					//to_chat(world, "DEBUG: Hit Chance 1: [hit_chance], Hit Roll: [hit_roll]")
 					#endif
-					if(hit_roll < 25 && !shot_from.sniper_target(A)) //Sniper targets more likely to hit
-						def_zone = pick(base_miss_chance)	// Still hit but now we might hit the wrong body part
-					if(!shot_from.sniper_target(A))
+					if(hit_roll < 25) //Sniper targets more likely to hit
+						if(shot_from && !shot_from.sniper_target(A)) //Avoid sentry run times
+							def_zone = pick(base_miss_chance)	// Still hit but now we might hit the wrong body part
+						else
+							def_zone = pick(base_miss_chance)	// Still hit but now we might hit the wrong body part
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/projectile.dm]: This `else` is doing the same as the `if` check, making it irrelevant.
Check the whole logic here, please.

**Pull request merged**